modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistryBootstrap.java

```diff
@@ -0,0 +1,489 @@
+ * parameter is specified. If no 'returnClass' is specified the default in the transformer will be used.
+ * <p/>
+ * Note that all objects defined have to have a default constructor. They can implement injection interfaces such as
+ * {@link MuleContextAware} and lifecycle interfaces such as {@link Initialisable}.
+ *
+ * @since 3.7.0
+ */
+public class SpringRegistryBootstrap implements MuleContextAware
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SpringRegistryBootstrap.class);
+    private static final String TRANSFORMER_KEY = ".transformer.";
+    private static final String OBJECT_KEY = ".object.";
+    private static final String SINGLE_TX = ".singletx.";
+
+    private final RegistryBootstrapDiscoverer discoverer;
+    private final BeanDefinitionRegistry beanDefinitionRegistry;
+
+    private ArtifactType supportedArtifactType = ArtifactType.APP;
+    private MuleContext muleContext;
+
+    public enum ArtifactType
+    {
+        APP("app"), DOMAIN("domain"), ALL("app/domain");
+
+        public static final String APPLY_TO_ARTIFACT_TYPE_PARAMETER_KEY = "applyToArtifactType";
+        private final String artifactTypeAsString;
+
+        ArtifactType(String artifactTypeAsString)
+        {
+            this.artifactTypeAsString = artifactTypeAsString;
+        }
+
+        public String getAsString()
+        {
+            return this.artifactTypeAsString;
+        }
+
+        public static ArtifactType createFromString(String artifactTypeAsString)
+        {
+            for (ArtifactType artifactType : values())
+            {
+                if (artifactType.artifactTypeAsString.equals(artifactTypeAsString))
+                {
+                    return artifactType;
+                }
+            }
+            throw new MuleRuntimeException(createStaticMessage("No artifact type found for value: " + artifactTypeAsString));
+        }
+    }
+
+    /**
+     * Creates a default SimpleRegistryBootstrap using a {@link ClassPathRegistryBootstrapDiscoverer}
+     * in order to get the Properties resources from the class path.
+     */
+    public SpringRegistryBootstrap(BeanDefinitionRegistry beanDefinitionRegistry)
+    {
+        this(new ClassPathRegistryBootstrapDiscoverer(), beanDefinitionRegistry);
+    }
+
+    /**
+     * Allows to specify a {@link RegistryBootstrapDiscoverer} to discover the Properties
+     * resources to be used.
+     * @param discoverer
+     */
+    public SpringRegistryBootstrap(RegistryBootstrapDiscoverer discoverer, BeanDefinitionRegistry beanDefinitionRegistry)
+    {
+        this.discoverer = discoverer;
+        this.beanDefinitionRegistry = beanDefinitionRegistry;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void setMuleContext(MuleContext context)
+    {
+        this.muleContext = context;
+    }
+
+    public void boot() throws BootstrapException
```

Comment:
this is a very similar class but has a lot of differences. This is also spring specific so I really think it's better for them not to know each other.